tests/cmd/fathom-elab.md

```diff
@@ -144,3 +144,22 @@ error: duplicate labels found in record
 
 
 ```
+
+### Module item cycle
+
+Cycles in a module are reported during elaboration
+
+```console
+$ fathom elab --module tests/fail/elaboration/item-cycle.fathom
+? failed
+error: cycle detected
+ = first → second → third → first
+
+error: cannot find `first` in scope
+  ┌─ tests/fail/elaboration/item-cycle.fathom:5:13
+  │
+5 │ def third = first;
+  │             ^^^^^ unbound name
```

Comment:
Curious why the body of this item is getting elaborated 🤔 - is the index of the item getting returned from `determine_order`?

Reply:
Yes it does: `visit_item` recurses into `first`s dependencies and encounters the cycle so it stops recursing. When that returns the index of `first` is added to the output since it's assumed that all its dependencies have been added before it.

https://github.com/yeslogic/fathom/pull/376/files#diff-936b5db84b3d389fc600527214d8189638f749750ff83bf3702c6379c19e7cadR130

Should it omit `first`s index if errors are detected visiting its dependencies?

Reply:
Yeah I think so - if the programmer wrote `first` they might get a bit frustrated if this was getting reported as being unbound. Not sure how hard it would be with the current algorithm, though!

Reply:
I think it's a pretty easy change. I'll give it a go

Reply:
Ok I was wrong about it being easy. I went around in circles for a bit but I think came up with a solution that handles it reasonably nicely in the end.

Reply:
Went around in circles you say,

Reply:
Indeed, but that connection was accidental.